src/components/EmbeddedWalletModal/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { Button, Icon, NavButton } from "@stellar/design-system";
+import { useCallback, useEffect } from "react";
+import { createPortal } from "react-dom";
+
+import "./styles.scss";
+
+interface EmbeddedWalletModalProps {
+  visible: boolean;
+  onClose?: () => void;
+  isDismissible?: boolean;
+  title: string;
+  description: string;
+  contentAlign?: "center" | "left";
+  primaryActionLabel?: string;
+  onPrimaryAction?: () => void;
+  isPrimaryActionLoading?: boolean;
+  isPrimaryActionDisabled?: boolean;
+}
+
+const MODAL_OPEN_CLASS_NAME = "modal-open";
+const MODAL_PARENT_ID = "embedded-wallet-modal-root";
+
+export const EmbeddedWalletModal = ({
+  visible,
+  onClose,
+  isDismissible = true,
+  title,
+  description,
+  contentAlign = "center",
+  primaryActionLabel,
+  onPrimaryAction,
+  isPrimaryActionLoading = false,
+  isPrimaryActionDisabled = false,
+}: EmbeddedWalletModalProps) => {
+  const isModalDismissible = Boolean(onClose) && isDismissible;
+
+  const handleClose = useCallback(() => {
+    onClose?.();
+  }, [onClose]);
+
+  const closeOnEscape = useCallback(
+    (event: KeyboardEvent) => {
+      if (!isModalDismissible) {
+        return;
+      }
+      if (event.key === "Escape") {
+        handleClose();
+      }
+    },
+    [handleClose, isModalDismissible],
+  );
+
+  useEffect(() => {
+    if (!visible || !isModalDismissible) {
+      return undefined;
+    }
+
+    document.addEventListener("keyup", closeOnEscape);
+    return () => {
+      document.removeEventListener("keyup", closeOnEscape);
+    };
+  }, [closeOnEscape, isModalDismissible, visible]);
+
+  useEffect(() => {
+    if (visible) {
+      document.body.classList.add(MODAL_OPEN_CLASS_NAME);
+      return () => {
+        document.body.classList.remove(MODAL_OPEN_CLASS_NAME);
+      };
+    }
+
+    document.body.classList.remove(MODAL_OPEN_CLASS_NAME);
+    return undefined;
+  }, [visible]);
```

Comment:
I'm not quite sure what we're trying to do here. Are we adding and removing a class based on visibility? 

`visible` block shouldn't return anything. The cleanup for `useEffect` should be a function.


```suggestion
  useEffect(() => {
    if (visible) {
      document.body.classList.add(MODAL_OPEN_CLASS_NAME)
    } else {
      document.body.classList.remove(MODAL_OPEN_CLASS_NAME);
    }
    
    return () => {
      document.body.classList.remove(MODAL_OPEN_CLASS_NAME);
    };
  }, [visible]);
```

Reply:
It was supposed to replicate this behaviour that prevents background scrolling while the modal is open, but it seems like it works without it, so I'll remove it https://github.com/stellar/stellar-design-system/blob/ab234b4b903b4b8e65008f7988d9ec2dff435265/%40stellar/design-system/src/components/Modal/index.tsx#L107-L113